### ISSUE TITLE
Unhandled Exception: System.AccessViolationException: Attempted to read or write protected memory. This is often an indication that other memory is corrupt.

### ISSUE THREAD

**Issue by iwanPlays** [mafaca/UtinyRipperFull#2]
Today's UtinyRipperFull x64
Initialize engine version: 5.3.5f1 (960ebf59018a) according to output_log.txt
Game: [Superhot ](https://store.steampowered.com/app/322500/SUPERHOT/)(current Steam version)

```
UtinyRipperFull.exe level0
Import: Dependency 'resources/unity_builtin_extra' was loaded
Import: Dependency 'Resources\unity default resources' was loaded
Import: Dependency 'globalgamemanagers.assets' was loaded
Import: Dependency 'sharedassets0.assets' was loaded
Import: Asset collection has versions probably incompatible with each other. Here they are:
Import: 5.3.5f1
Import: 5.3.4p1
Export: 'ScreenSelector gold same krysztalki(Texture2D)' exported
Export: 'SUPERHOTdefaultPhysicMaterial(PhysicMaterial)' exported
Export: 'Default-Skybox(Material)' exported

Unhandled Exception: System.AccessViolationException: Attempted to read or write protected memory. This is often an indication that other memory is corrupt.
   at DotNetDxc.IDxcLibrary.GetBlobAstUf16(IDxcBlob blob)
   at UtinyRipperFull.Exporters.ShaderDXExporter.GetStringFromBlob(IDxcLibrary library, IDxcBlob blob)
   at UtinyRipperFull.Exporters.ShaderDXExporter.Export(Byte[] shaderData, TextWriter writer)
   at UtinyRipper.Classes.Shaders.ShaderSubProgram.Export(TextWriter writer, Func`2 exporterInstantiator)
   at UtinyRipper.Classes.Shaders.ShaderSubProgramBlob.Export(TextWriter writer, String header, Func`2 exporterInstantiator)
   at UtinyRipper.Classes.Shader.ExportBinary(IExportContainer container, Stream stream, Func`2 exporterInstantiator)
   at UtinyRipperFull.Exporters.ShaderAssetExporter.Export(IExportContainer container, Object asset, String path)
   at UtinyRipper.AssetExporters.AssetExportCollection.ExportInner(ProjectAssetContainer container, String filePath)
   at UtinyRipper.AssetExporters.AssetExportCollection.Export(ProjectAssetContainer container, String dirPath)
   at UtinyRipper.AssetExporters.ProjectExporter.Export(String path, IEnumerable`1 assets)
   at UtinyRipperFull.Program.Load(IReadOnlyList`1 args)
   at UtinyRipperFull.Program.Main(String[] args)
```

Of course it could be just my RAM being broken.

After reading https://stackoverflow.com/questions/4074585/attempted-to-read-or-write-protected-memory-this-is-often-an-indication-that-ot , I tried UtinyRipper x86 - same problem.

**Comment by mafaca** [comment]
It happens because shader has unsupported format. I'll try to fix it later

**Comment by mafaca** [comment]
I've had familiar issue with other file. I temporarily fixed it in 0082eff7bb7d44411d8b93580d799474aafcf5bc. I think it may also help in your case.

**Comment by iwanPlays** [comment]
Thanks! level0 now finishes ripping, level1 doesn't cause this error and reaches a different error reported in https://github.com/mafaca/UtinyRipper/issues/30
